### PR TITLE
WEB-155 inject version number from github sdk releases

### PR DESCRIPTION
**Pull request by gregsadetsky**
done with a custom plugin! :-)

markdown code:
image

web output:
image

### CODE REVIEW

**Comment by kochis** [comment]
Do you want to update all the occurrences with the new script, to make sure they work?

```
$ ack 'js.radar.com'

docs/maps/maps.mdx
88:    <link href="https://js.radar.com/v4.1.11/radar.css" rel="stylesheet">
89:    <script src="https://js.radar.com/v4.1.11/radar.min.js"></script>

docs/maps/autocomplete.mdx
62:    <link href="https://js.radar.com/v4.1.6/radar.css" rel="stylesheet">
63:    <script src="https://js.radar.com/v4.1.6/radar.min.js"></script>

docs/sdk/web.mdx
17:<link href="https://js.radar.com/v4.1.18/radar.css" rel="stylesheet">
18:<script src="https://js.radar.com/v4.1.18/radar.min.js"></script>

docs/tutorials/localizing-a-website-based-on-current-country-or-city.mdx
37:<script src="https://js.radar.com/v3/radar.min.js"></script>
80:    <script src="https://js.radar.com/v3/radar.min.js"></script>

docs/tutorials/building-an-on-premise-mode.mdx
84:    <script src="https://js.radar.com/v3/radar.min.js"></script>
210:    <script src="https://js.radar.com/v3/radar.min.js"></script>
363:    <script src="https://js.radar.com/v3/radar.min.js"></script>
```

The ones that just say `v3` can be updated to just `v4` (that always points to latest).

**Comment by gregsadetsky** [comment]
For sure! I didn't want to overcomplicate it, but I did consider adding `$RADAR_(PLATFORM)_SDK_VERSION_MAJOR$`, `_MINOR` and `_PATCH` too (e.g. v4, v4.18, v4.18.4) -- I saw that some of the native sdk examples do sometimes include only major+minor, not the whole version.

In any case, I could do the "v4"'s like that automatically or manually edit them in. Let me know if this makes sense.

**Comment by kochis** [comment]
Probably don't need to automate the major versions, as they don't update that frequently.

I would assume we'd probably want to always show the full version (major, minor, patch) where the version is specified.

File: docusaurus.config.js [hunk 1]
```diff
@@ -71,6 +73,34 @@ module.exports = {
...
+    // docusaurus. declaring the plugin "inline" below works.
+    () => ({
+        // name is crucial here for CodeBlock/index.js to get this plugin data
+        name: 'radar-version-injector',
+        async contentLoaded({content, actions}) {
+            const {setGlobalData} = actions;
+            const RADAR_SDK_REPOS = {
+                'web': 'radarlabs/radar-sdk-js',
+                'ios': 'radarlabs/radar-sdk-ios',
+                'android': 'radarlabs/radar-sdk-android',
```
Comment (kochis) [inline docusaurus.config.js:89]: there's also react-native, and capacitor (if we have links for those)

**Comment by gregsadetsky** [comment]
> Do you want to update all the occurrences with the new script, to make sure they work?
> 
[...]
> 
> The ones that just say `v3` can be updated to just `v4` (that always points to latest).

done for all of the occurrences of js.radar.com + updated v3 to v4 - bc06c9590ba9d8c0c84a092355908df1272b7093

File: docusaurus.config.js [hunk 2]
```diff
@@ -71,6 +73,85 @@ module.exports = {
...
+                        fetch(`https://api.github.com/repos/radarlabs/${repo}/releases/latest`),
+                        // delay between each api call to avoid rate limiting
+                        new Promise(resolve => setTimeout(resolve, 1000))
+                    ]);
+                    // check that response status code is 200, otherwise throw / stop immediately
+                    if (response.status !== 200) {
+                        throw new Error(`An error occurred while fetching versions from GitHub! ${response.status} -- ${response.statusText}`);
+                    }
+                    const data = await response.json();
+                    radarGitHubReleases[`RADAR_${sdkName.toUpperCase()}_SDK_VERSION`] = data.name;
```
Comment (kochis) [inline docusaurus.config.js:136]: If any of this fails, does it stop the build?

Comment (gregsadetsky) [reply docusaurus.config.js]: yes, that was my original intent i.e. better to break than not have incorrect versions :-) but that was probably a bit too harsh

just added fallback versions for all SDKs in fc489b4c -- we're back to hardcoded versions after all :-) but these will only be used if any of the github api calls fail.

the only remaining "hard" build stop [is this](https://github.com/radarlabs/documentation/pull/455/files#diff-a038096cbdea434999e1dce5ab497212f1fe18204dde1a027ce3bdd663261a2aR154) -- i.e. if, after trying to fetch the version number from github AND after (supposedly) using a fallback version number, the web sdk version is still blank, then the build will fail as a last resort sanity check/validation.

Comment (kochis) [reply docusaurus.config.js]: No, that's fine. Was just confirming that was the behavior.

**Pull request merged**